File: src/main/java/io/legado/app/utils/TextUtils.java
package io.legado.app.utils;

import java.util.Iterator;

public class TextUtils {
   public static boolean isEmpty(CharSequence str) {
      return str == null || str.length() == 0;
   }

   public static String join(CharSequence delimiter, Object[] tokens) {
      int length = tokens.length;
      if (length == 0) {
         return "";
      } else {
         StringBuilder sb = new StringBuilder();
         sb.append(tokens[0]);

         for(int i = 1; i < length; ++i) {
            sb.append(delimiter);
            sb.append(tokens[i]);
         }

         return sb.toString();
      }
   }

   public static String join(CharSequence delimiter, Iterable tokens) {
      Iterator<?> it = tokens.iterator();
      if (!it.hasNext()) {
         return "";
      } else {
         StringBuilder sb = new StringBuilder();
         sb.append(it.next());

         while(it.hasNext()) {
            sb.append(delimiter);
            sb.append(it.next());
         }

         return sb.toString();
      }
   }
}

File: src/main/kotlin/io/legado/app/utils/TextUtils.kt
package io.legado.app.utils

object TextUtils {
    @JvmStatic fun isEmpty(value: CharSequence?): Boolean = value == null || value.isEmpty()
    @JvmStatic fun join(delimiter: CharSequence, tokens: Array<out Any?>): String = tokens.joinToString(delimiter)
    @JvmStatic fun join(delimiter: CharSequence, tokens: Iterable<*>): String = tokens.joinToString(delimiter)
}
